### PR TITLE
[G2M] dateRange 

### PR DESCRIPTION
**Pull request by kc-leung**
- Add date range component for lumen-table

### CODE REVIEW

File: src/components/date-range/index.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    <form ref={ref} className={`${dateRangeClasses.form}`} noValidate>
+      <div className={`fieldContainer ${dateRangeClasses.field}`}>
+        {showLabel && <label className={`${dateRangeClasses.label}`} htmlFor="from">From</label>}
+        <input className={`date-input ${dateRangeClasses.input}`} type="date" id="from" name="from"
+          value={defaultValue[0] || ''} onChange={({ target: { value } }) => setFromValue(value)}/>
+      </div>
+      <div className={`fieldContainer ${dateRangeClasses.field}`}>
+        {showLabel && <label className={`${dateRangeClasses.label}`} htmlFor="to">To</label>}
+        <input className={`date-input ${dateRangeClasses.input}`} type="date" id="to" name="to"
+          value={defaultValue[1] || ''} onChange={({ target: { value } }) => setToValue(value)}/>
```
Comment (woozyking) [inline src/components/date-range/index.js:25]: it may be worth documenting the `value` here in the form of `yyyy-mm-dd`, which may be different from what the user sees in the display which goes by the browser locale setting

see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date#handling_browser_support for more context and perhaps confine it using the `pattern` attribute

**Review by woozyking — APPROVED** [review]
looks pretty good. see comments for minor concern

also let's try to extend this to single date/date-time/time selectors, and date-time-range selector for comprehensiveness (future iterations on this path)

also timezone support etc

**Review by sallkall — APPROVED** [review]

File: src/components/date-range/date-range.css [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.date-input:hover {
+  border-color: black;
+  border-bottom-width: 2px;
+}
```
Comment (hyx131) [inline src/components/date-range/date-range.css:4]: this could be replaced by `tailwindcss` with: `'hover:border-back hover:border-b-2'`

**Review by hyx131 — APPROVED** [review]

File: src/components/date-range/date-range.css (hunk 2 above)
Comment (kc-leung) [reply src/components/date-range/date-range.css:4]: doesn't work with border width :( I added hover for the borderWidth attribute and still doesn't work. So I decided to have both together although I can move border-color color out that one works

Comment (hyx131) [reply src/components/date-range/date-range.css:3]: 🤔  did you add to the `variants` section?
i added to the `variants` section this: `borderWidth: ['responsive', 'hover']` and it seems to be working on my side

Comment (kc-leung) [reply src/components/date-range/date-range.css:3]: yea I did let me try again maybe I forgot to save lol

Comment (hyx131) [reply src/components/date-range/date-range.css:3]: oki lol, and don't forget to run `yarn build-tailwindcss` to re-build the css after you saved the config :)

Comment (kc-leung) [reply src/components/date-range/date-range.css:3]: that was the setup I was missing hehehe thanks~! it works now
